packages/@react-aria/select/src/HiddenSelect.tsx

```diff
@@ -132,6 +132,16 @@ export function HiddenSelect<T>(props: HiddenSelectProps<T>) {
   let selectRef = useRef(null);
   let {containerProps, inputProps, selectProps} = useHiddenSelect({...props, selectRef}, state, triggerRef);
 
+  // When the selected key changes, we dispatch a change event to mimic the behavior of a native select element.
+  let prevSelectedKey = useRef(state.selectedKey);
```

Comment:
The issue with this one is that it fires a change event for a form `reset`, which isn't how native works.

We might be able to get around it by doing something like

```
  let moreThan300Ref = useRef(null);
  let lessThan300Ref = useRef(null);
  let wasReset = useRef(false);
  useEffect(() => {
    let onFormReset = () => {
      wasReset.current = true;
    };
    let form = moreThan300Ref.current?.form || lessThan300Ref.current?.form;
    if (form) {
      form.addEventListener('reset', onFormReset);
      return () => {
        form.removeEventListener('reset', onFormReset);
      };
    }
  }, [state.collection.size]);
...
  let prevSelectedKey = useRef(state.selectedKey);
  useEffect(() => {
    if (wasReset.current) {
      wasReset.current = false;
      return;
    }
    if (prevSelectedKey.current === state.selectedKey || !selectRef.current) {
      return;
    }
...
```

Though I'm not sure if there are other missing cases, such as controlled Pickers.